packages/compass-import-export/src/components/import-modal.tsx

```diff
@@ -216,18 +255,40 @@ function ImportModal({
           ignoreBlanks={ignoreBlanks}
           setIgnoreBlanks={setIgnoreBlanks}
         />
-        {fileType === 'csv' && (
+        {fileType === 'csv' && csvAnalyzed && (
           <FormFieldContainer>
             <ImportPreview
               loaded={previewLoaded}
-              analyzed={csvAnalyzed}
               onFieldCheckedChanged={toggleIncludeField}
               setFieldType={setFieldType}
               values={values}
               fields={fields as FieldFromCSV[]}
             />
           </FormFieldContainer>
         )}
+
+        {fileType === 'csv' && !csvAnalyzed && (
+          <FormFieldContainer className={progressStyles}>
+            <Body weight="medium">Detecting field types</Body>
+            {analyzeBytesTotal && (
+              <div className={loaderStyles}>
+                <SpinLoader />
+                <Body>
+                  {Math.round(
+                    (analyzeBytesProcessed / analyzeBytesTotal) * 100
+                  )}
```

Comment:
Maybe also something to keep in mind for the future as I'm guessing you are just following a pattern here. The only place where these numbers are used is this placeholder, so following [redux design guidelines](https://redux.js.org/style-guide/#connect-more-components-to-read-data-from-the-store), you might want to rather have a separate "analyze in-progress" component, which will be the only one that subscribes to these values via connect, not the whole modal